File: src-tauri/src/commands/warmup.rs
use crate::error::AppResult;
use crate::models::warmup::{AgentConfig, RunLog, TimePreset, TriggerType};
use crate::services::warmup_service::WarmupState;

#[tauri::command]
pub async fn warmup_list_agents(state: tauri::State<'_, WarmupState>) -> AppResult<Vec<AgentConfig>> {
    Ok(state.list_agents().await)
}

#[tauri::command]
pub async fn warmup_save_agent(
    state: tauri::State<'_, WarmupState>,
    agent: AgentConfig,
) -> AppResult<AgentConfig> {
    state.save_agent(agent).await
}

#[tauri::command]
pub async fn warmup_delete_agent(state: tauri::State<'_, WarmupState>, id: String) -> AppResult<()> {
    state.delete_agent(id).await
}

#[tauri::command]
pub async fn warmup_trigger_agent(
    state: tauri::State<'_, WarmupState>,
    agent_id: String,
    trigger_type: TriggerType,
) -> AppResult<RunLog> {
    state.trigger_agent(agent_id, trigger_type).await
}

#[tauri::command]
pub async fn warmup_list_logs(state: tauri::State<'_, WarmupState>) -> AppResult<Vec<RunLog>> {
    Ok(state.list_logs().await)
}

#[tauri::command]
pub async fn warmup_list_presets(state: tauri::State<'_, WarmupState>) -> AppResult<Vec<TimePreset>> {
    Ok(state.list_presets().await)
}

#[tauri::command]
pub async fn warmup_save_preset(
    state: tauri::State<'_, WarmupState>,
    preset: TimePreset,
) -> AppResult<TimePreset> {
    state.save_preset(preset).await
}

#[tauri::command]
pub async fn warmup_delete_preset(state: tauri::State<'_, WarmupState>, id: String) -> AppResult<()> {
    state.delete_preset(id).await
}

File: src/lib/tauri/commands/warmup.ts
import { invoke } from "@tauri-apps/api/core";

export type AgentKind = "claude" | "codex" | "custom";
export type ScheduleKind = "daily" | "weekly";
export type TriggerType = "manual" | "warmup_5h" | "weekly";
export type RunStatus =
  | "running"
  | "success"
  | "failed"
  | "timeout"
  | "skipped";
export type LogClearScope = "all" | "older_than_30_days" | "older_than_7_days";

export interface EnvVar {
  key: string;
  value: string;
}

export interface Schedule {
  id: string;
  enabled: boolean;
  kind: ScheduleKind;
  triggerType: TriggerType;
  time: string;
  daysOfWeek: number[];
  lastTriggeredAt?: string | null;
}

export interface AgentConfig {
  id: string;
  name: string;
  kind: AgentKind;
  enabled: boolean;
  command: string;
  args: string[];
  cwd?: string | null;
  env: EnvVar[];
  timeoutSeconds: number;
  schedules: Schedule[];
  createdAt: string;
  updatedAt: string;
}

export interface TimePreset {
  id: string;
  label: string;
  kind: ScheduleKind;
  time: string;
  daysOfWeek: number[];
  lastUsedAt: string;
}

export interface RunLog {
  id: string;
  agentId: string;
  agentName: string;
  triggerType: TriggerType;
  startedAt: string;
  finishedAt?: string | null;
  command: string;
  args: string[];
  cwd?: string | null;
  exitCode?: number | null;
  status: RunStatus;
  stdout: string;
  stderr: string;
  errorMessage?: string | null;
  durationMs?: number | null;
}

export const warmupCommands = {
  listAgents: () => invoke<AgentConfig[]>("warmup_list_agents"),
  saveAgent: (agent: AgentConfig) =>
    invoke<AgentConfig>("warmup_save_agent", { agent }),
  deleteAgent: (id: string) => invoke<void>("warmup_delete_agent", { id }),
  triggerAgent: (agentId: string, triggerType: TriggerType) =>
    invoke<RunLog>("warmup_trigger_agent", { agentId, triggerType }),
  listLogs: () => invoke<RunLog[]>("warmup_list_logs"),
  clearLogs: (scope: LogClearScope) =>
    invoke<number>("warmup_clear_logs", { scope }),
  listPresets: () => invoke<TimePreset[]>("warmup_list_presets"),
  savePreset: (preset: TimePreset) =>
    invoke<TimePreset>("warmup_save_preset", { preset }),
  deletePreset: (id: string) => invoke<void>("warmup_delete_preset", { id }),
};
